pkg/openthread/contrib/platform_radio.c

```diff
@@ -0,0 +1,422 @@
+    sReceiveFrame.mLength = len + RADIO_IEEE802154_FCS_LEN;
+    sReceiveFrame.mPower = _get_power();
+
+    /* Read received frame */
+    int res = dev->driver->recv(dev, (char *) sReceiveFrame.mPsdu, len, NULL);
+
+   DEBUG("Received message: len %d\n", (int) sReceiveFrame.mLength);
+    for (int i = 0; i < sReceiveFrame.mLength; ++i)
+    {
+        DEBUG("%x ", sReceiveFrame.mPsdu[i]);
+    }
+    DEBUG("\n");
+
+    /* Tell OpenThread that receive has finished */
+    otPlatRadioReceiveDone(aInstance, res > 0 ? &sReceiveFrame : NULL, res > 0 ? kThreadError_None : kThreadError_Abort);
+}
+
+/* Called upon TX event */
+void send_pkt(otInstance *aInstance, netdev_t *dev, netdev_event_t event)
+{
+    /* Tell OpenThread transmission is done depending on the NETDEV event */
+    switch (event) {
+        case NETDEV_EVENT_TX_COMPLETE:
+            DEBUG("openthread: NETDEV_EVENT_TX_COMPLETE\n");
+            otPlatRadioTransmitDone(aInstance, &sTransmitFrame, false, kThreadError_None);
+            break;
+        case NETDEV_EVENT_TX_COMPLETE_DATA_PENDING:
+            DEBUG("openthread: NETDEV_EVENT_TX_COMPLETE_DATA_PENDING\n");
+            otPlatRadioTransmitDone(aInstance, &sTransmitFrame, true, kThreadError_None);
+            break;
+        case NETDEV_EVENT_TX_NOACK:
+            DEBUG("openthread: NETDEV_EVENT_TX_NOACK\n");
+            otPlatRadioTransmitDone(aInstance, &sTransmitFrame, false, kThreadError_NoAck);
+            break;
+        case NETDEV_EVENT_TX_MEDIUM_BUSY:
+            DEBUG("openthread: NETDEV_EVENT_TX_MEDIUM_BUSY\n");
+            otPlatRadioTransmitDone(aInstance, &sTransmitFrame, false, kThreadError_ChannelAccessFailure);
+            break;
+        default:
+            break;
+    }
+}
+
+/* OpenThread will call this for setting PAN ID */
+void otPlatRadioSetPanId(otInstance *aInstance, uint16_t panid)
+{
+    DEBUG("openthread: otPlatRadioSetPanId: setting PAN ID to %04x\n", panid);
+    _set_panid(panid);
+}
+
+/* OpenThread will call this for setting extended address */
+void otPlatRadioSetExtendedAddress(otInstance *aInstance, uint8_t *aExtendedAddress)
+{
+    DEBUG("openthread: otPlatRadioSetExtendedAddress\n");
+    uint8_t reversed_addr[IEEE802154_LONG_ADDRESS_LEN];
+    for (int i = 0; i < IEEE802154_LONG_ADDRESS_LEN; i++) {
+        reversed_addr[i] = aExtendedAddress[IEEE802154_LONG_ADDRESS_LEN - 1 - i];
+    }
+    _set_long_addr(reversed_addr);
+}
+
+/* OpenThread will call this for setting short address */
+void otPlatRadioSetShortAddress(otInstance *aInstance, uint16_t aShortAddress)
+{
+    DEBUG("openthread: otPlatRadioSetShortAddress: setting address to %04x\n", aShortAddress);
+    _set_addr(((aShortAddress & 0xff) << 8) | ((aShortAddress >> 8) & 0xff));
+}
+
+/* OpenThread will call this for enabling the radio */
+ThreadError otPlatRadioEnable(otInstance *aInstance)
+{
+    DEBUG("openthread: otPlatRadioEnable\n");
+    (void) aInstance;
+
+    ThreadError error;
+
+    if (sDisabled) {
```

Comment:
the sDisabled var is needed because OpenThread asks the radio state in other functions (and should be consistent).

I think it's a good idea to put the transceiver in sleep if sDisabled = false. I will do it.
Cheers!